### PR TITLE
Feature: snapshot and restore scripts

### PR DESCRIPTION
**Pull request by lsh-0**
* adds a README with plain language instructions for creating and restoring ES snapshots
* adds create and restore snapshot scripts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+isn't that special, it's just a directory on the filesystem.
+
+Creating repositories and snapshots is easy. Subsequent snapshots in the same repository are incremental and will only 
+capture the changes made since the last snapshot.
+
+### creating repositories
+
+Before a snapshot can exist it must be created in a place ES knows about called a 'repository'. `fs` type repositories 
+exist on the filesystem and ES needs to know their location as well with the setting `path.repo` in 
+`/etc/elasticsearch/elasticsearch.yml`.
```
Comment (giorgiosironi) [inline README.md]: This is taken care of by the formula? Would state that if it is the case.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+To create a repository issue this api call: `PUT /_snapshot/{repo}` with the payload:
+
+```json
+{"type": "fs", "settings": {"location": "$repo"}}
+```
+
+Where `$repo` is the name of a path rooted in the `path.repo` setting (multiple paths may exist).
+
+[Reference](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/modules-snapshots.html#_repositories)
```
Comment (giorgiosironi) [inline README.md]: I have a feeling we are on an outdated version, unrelated

File: salt/search/scripts/create-snapshot.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+function create_snapshot {
+    # snapshots are incremental, so just the difference from the previous snapshot are stored
+    errcho "creating snapshot with '$snapshot' (blocking call, may take a while depending on size of snapshot)"
+    curlit -XPUT "$elasticsearch/_snapshot/$repo/$snapshot?wait_for_completion=true"
+}
+
+function compress_snapshot {
+    errcho "compressing snapshot at '$repo_path'"
+    output_fname="$snapshot.tar.gz"
+    tar -czf "$output_fname" -C "$repo_path/.." "$repo"
```
Comment (giorgiosironi) [inline salt/search/scripts/create-snapshot.sh]: I don't thin this script will need `sudo`, but if so could make the permissions 744

File: README.md (hunk 2 above)
Comment (lsh-0) [reply README.md]: (we are ;)

but for now the PPA works and after the 18.04 upgrade, if we're game, we can look at upgrading to ES 7

File: salt/search/scripts/create-snapshot.sh (hunk 3 above)
Comment (lsh-0) [reply salt/search/scripts/create-snapshot.sh]: the script lives in /root , it's for administration and ad-hoc use only. I might be able to incorporate it into a regular ubr backup in the future

File: README.md (hunk 1 above)
Comment (lsh-0) [reply README.md]: done, made it clearer

**Pull request merged**